android/app/src/main/java/app/organicmaps/widget/menu/NavMenu.java

```diff
@@ -208,21 +209,20 @@ private void updateTimeEstimate(int seconds)
 
   private void updateSpeedView(@NonNull RoutingInfo info)
   {
-    final Location last = LocationHelper.from(mActivity).getSavedLocation();
-    if (last == null)
-      return;
-
-    Pair<String, String> speedAndUnits = StringUtils.nativeFormatSpeedAndUnits(last.getSpeed());
+    SpeedFormatted speed = info.speed;
+    boolean speedLimitExceeded = false;
 
-    if (info.speedLimitMps > 0.0)
+    if (info.speedLimit != null && info.speedLimit.isValid())
     {
-      Pair<String, String> speedLimitAndUnits = StringUtils.nativeFormatSpeedAndUnits(info.speedLimitMps);
-      mSpeedValue.setText(speedAndUnits.first + "\u202F/\u202F" + speedLimitAndUnits.first);
+      SpeedFormatted speedLimit = info.speedLimit;
+      mSpeedValue.setText(speed.mSpeedStr + "\u202F/\u202F" + speedLimit.getUnitsStr(mActivity));
+
+      speedLimitExceeded = (int)speed.mSpeed > (int)speedLimit.mSpeed;
```

Comment:
Here we decide if there was speeding

Reply:
Here we compare km with km, or mi with mi, right? What about the threshold idea?